### DISCUSSION TITLE
Is there a way to make wavesurfer draggable?

### DISCUSSION THREAD

**Discussion by SoplicaSoftware** [katspaugh/wavesurfer.js#3129] [Q&A] [answered]
Hi, I'm trying to implement drag and drop feature to my player that utilizes wavesurfer.js in electron.js. When user drags wavesurfer to deskstop it needs to get currently played file there. I'm using draggable="true" on wavesurver container and events from official docs of  electrons native drag and drop, but that does not seem to work.

**Comment by katspaugh** [comment]
You should read up on [HTML drag-n-drop](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API).

In a nutshell, `draggable="true"` is for native browser drag-n-drop for things like images and media elements. Since wavesurfer is neither, you should use drag events to make it work the way you want. It's probably best to attach these event handlers to your container element.

Let me know if you have further questions.

**Reply by SoplicaSoftware** [reply]
unfortunately i think, that the wavesurfer container can't trigger drag events when dragged. Maybe because its not the container that I drag, but its the wavesurfer itself? Any other div triggers the event.

**Reply by katspaugh** [reply]
Make sure you're attaching it to _your_ container div, not to one of the elements wavesurfer creates. You might have to use `{ capture: true }` in the event options for it to trigger on the container and not on the inner elements. See https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#event_capture.

**Reply by SoplicaSoftware** [reply]
However, I'm still facing difficulty triggering the drag event. It seems that any user mouse input within the wavesurfer container area is being exclusively captured by wavesurfer itself and doesn't trigger the container's events. I'm beginning to consider that my approach might be fundamentally flawed. This is due to the fact that even if I managed to initiate the drag event, it would inevitably result in an undesired interaction with the player.

**Reply by katspaugh** [reply]
You're right, native drag-n-drop is being intercepted by wavesurfer's elements no matter what. I've made [this example](https://wavesurfer-js.org/examples/#698c67c94a678af22ad3df77fd12ca85) that has `{ capture: true }` but it doesn't help.

I suspect it has to do with `e.preventDefault` on pointerdown and pointermove in [draggable.ts](https://github.com/katspaugh/wavesurfer.js/blob/main/src/draggable.ts#L27).

**Reply by SoplicaSoftware** [reply]
So is there a way to fix this? Or is this impossible in the current state of wavesurfer?

**Reply by katspaugh** [reply]
Try removing `e.preventDefault` in draggable.ts.

**Reply by SoplicaSoftware** [reply]
Where should the draggable.ts file located? I only have draggable.js and draggable.d.ts. draggable.js has multiple `e.preventDefault`.

Forgot to mention, package was downloaded using npm.

**Reply by katspaugh** [reply]
I'm talking about running this repo locally.

**Reply by katspaugh** [reply, marked as answer]
@SoplicaIndustries this is now fixed in 7.2.0 (currently in beta). See https://wavesurfer-js.org/examples/#698c67c94a678af22ad3df77fd12ca85

**Reply by SoplicaSoftware** [reply]
Thanks a lot!
